File: api/submit-lead.js
const https = require('https');
const fs = require('fs');
const path = require('path');

module.exports = async (req, res) => {
  // Set CORS headers
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'POST, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  // Handle preflight
  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }

  // Only allow POST
  if (req.method !== 'POST') {
    return res.status(405).json({ success: false, error: 'Method not allowed' });
  }

  try {
    const { name, email, phone, service, message } = req.body;

    // Validate required fields
    if (!name || !email || !phone || !service || !message) {
      return res.status(400).json({ 
        success: false, 
        error: 'All fields are required' 
      });
    }

    // Prepare HCP lead data
    const hcpData = JSON.stringify({
      customer: {
        first_name: name.split(' ')[0] || name,
        last_name: name.split(' ').slice(1).join(' ') || '',
        email: email,
        mobile_number: phone,
        notifications_enabled: true
      },
      source: 'Website - c4groundcontrol.com',
      service_type: service,
      description: `Service Interest: ${service}\n\nCustomer Message:\n${message}`,
      tags: [service, 'Website Lead']
    });

    // Create lead in HousecallPro
    const hcpResponse = await new Promise((resolve, reject) => {
      const options = {
        hostname: 'api.housecallpro.com',
        path: '/leads',
        method: 'POST',
        headers: {
          'Authorization': 'Token 10d6d33f386e4d0d9eba59e89313c53d',
          'Content-Type': 'application/json',
          'Content-Length': Buffer.byteLength(hcpData)
        }
      };

      const hcpReq = https.request(options, (hcpRes) => {
        let data = '';
        
        hcpRes.on('data', (chunk) => {
          data += chunk;
        });
        
        hcpRes.on('end', () => {
          if (hcpRes.statusCode >= 200 && hcpRes.statusCode < 300) {
            resolve({ success: true, data: JSON.parse(data) });
          } else {
            reject(new Error(`HCP API error: ${hcpRes.statusCode} - ${data}`));
          }
        });
      });

      hcpReq.on('error', (error) => {
        reject(error);
      });

      hcpReq.write(hcpData);
      hcpReq.end();
    });

    // Save backup locally (Note: Vercel serverless functions have /tmp directory)
    const timestamp = new Date().toISOString().replace(/[:.]/g, '-');
    const filename = `${name.replace(/\s+/g, '-')}-${timestamp}.json`;
    const backupDir = '/tmp/website-leads'; // Vercel temp directory
    
    // For local development, use the home directory path
    const isLocal = process.env.NODE_ENV !== 'production';
    const localBackupDir = path.join(process.env.HOME || '/Users/c4groundcontrol', 'Documents', 'C4-AI-Bot', 'Website-Leads');
    
    const targetDir = isLocal ? localBackupDir : backupDir;
    
    try {
      // Create directory if it doesn't exist
      if (!fs.existsSync(targetDir)) {
        fs.mkdirSync(targetDir, { recursive: true });
      }

      const backupData = {
        timestamp: new Date().toISOString(),
        formData: { name, email, phone, service, message },
        hcpResponse: hcpResponse.data
      };

      fs.writeFileSync(
        path.join(targetDir, filename),
        JSON.stringify(backupData, null, 2)
      );
    } catch (backupError) {
      console.error('Backup save error:', backupError);
      // Don't fail the request if backup fails
    }

    // Return success
    return res.status(200).json({
      success: true,
      message: 'Lead submitted successfully',
      customerId: hcpResponse.data.id || null
    });

  } catch (error) {
    console.error('Error processing lead:', error);
    return res.status(500).json({
      success: false,
      error: 'Failed to submit lead. Please try again or call us directly.',
      details: error.message
    });
  }
};

File: api/submit-lead.py
#!/usr/bin/env python3
"""
C4 Ground Control Lead Capture API
Receives form submissions from website → creates HCP customers automatically
"""

import json
import sys
import urllib.request
import urllib.parse
from datetime import datetime

# HCP API Configuration
HCP_API_KEY = "10d6d33f386e4d0d9eba59e89313c53d"
HCP_API_BASE = "https://api.housecallpro.com"

def create_hcp_customer(form_data):
    """Create customer in HousecallPro"""
    
    # Prepare customer data
    customer_payload = {
        "first_name": form_data.get("firstName", ""),
        "last_name": form_data.get("lastName", ""),
        "email": form_data.get("email", ""),
        "mobile_number": form_data.get("phone", ""),
        "company": form_data.get("company", ""),
        "address": {
            "street": form_data.get("address", ""),
            "city": "",
            "state": "FL",
            "zip": "",
            "country": "US"
        },
        "lead_source": "Website - c4groundcontrol.com"
    }
    
    # Create customer
    headers = {
        "Authorization": f"Bearer {HCP_API_KEY}",
        "Content-Type": "application/json"
    }
    
    req = urllib.request.Request(
        f"{HCP_API_BASE}/customers",
        data=json.dumps(customer_payload).encode('utf-8'),
        headers=headers,
        method='POST'
    )
    
    try:
        with urllib.request.urlopen(req, timeout=10) as response:
            customer_data = json.loads(response.read().decode('utf-8'))
            customer_id = customer_data.get("customer", {}).get("id")
            
            # Create estimate with project details
            if customer_id:
                create_estimate(customer_id, form_data)
            
            return {
                "success": True,
                "customer_id": customer_id,
                "customer_data": customer_data
            }
            
    except urllib.error.HTTPError as e:
        error_body = e.read().decode('utf-8')
        return {
            "success": False,
            "error": f"HCP API error: {e.code} - {error_body}"
        }
    except Exception as e:
        return {
            "success": False,
            "error": str(e)
        }

def create_estimate(customer_id, form_data):
    """Create estimate in HCP with project details"""
    
    property_type = form_data.get("propertyType", "")
    services = form_data.get("services", "")
    message = form_data.get("message", "")
    
    # Build estimate description
    description = f"""
Lead from Website - {datetime.now().strftime('%Y-%m-%d %H:%M')}

Property Type: {property_type}
Services Requested: {services}

Customer Notes:
{message}

FOLLOW UP REQUIRED - Schedule site visit for quote
    """.strip()
    
    estimate_payload = {
        "customer_id": customer_id,
        "options": [
            {
                "name": f"{services} - Initial Consultation",
                "line_items": [
                    {
                        "name": "Site Visit & Consultation",
                        "description": description,
                        "quantity": 1,
                        "unit_price": 0,  # Free consultation
                        "unit_cost": 0
                    }
                ]
            }
        ]
    }
    
    headers = {
        "Authorization": f"Bearer {HCP_API_KEY}",
        "Content-Type": "application/json"
    }
    
    req = urllib.request.Request(
        f"{HCP_API_BASE}/estimates",
        data=json.dumps(estimate_payload).encode('utf-8'),
        headers=headers,
        method='POST'
    )
    
    try:
        with urllib.request.urlopen(req, timeout=10) as response:
            estimate_data = json.loads(response.read().decode('utf-8'))
            return estimate_data
    except Exception as e:
        print(f"Warning: Could not create estimate - {e}", file=sys.stderr)
        return None

def save_lead_backup(form_data, hcp_result):
    """Save lead to local backup file"""
    import os
    
    backup_dir = os.path.expanduser("~/Documents/C4-AI-Bot/Website-Leads")
    os.makedirs(backup_dir, exist_ok=True)
    
    timestamp = datetime.now().strftime("%Y%m%d-%H%M%S")
    filename = f"{backup_dir}/{timestamp}-{form_data.get('lastName', 'Unknown')}.json"
    
    lead_record = {
        "timestamp": datetime.now().isoformat(),
        "form_data": form_data,
        "hcp_result": hcp_result
    }
    
    with open(filename, 'w') as f:
        json.dump(lead_record, f, indent=2)
    
    return filename

def main():
    """Main API endpoint handler"""
    
    # Read form data from stdin (sent by web server)
    try:
        form_data = json.load(sys.stdin)
    except json.JSONDecodeError:
        print(json.dumps({
            "success": False,
            "error": "Invalid JSON input"
        }))
        return 1
    
    # Validate required fields
    required_fields = ["firstName", "lastName", "phone", "address", "services"]
    missing_fields = [f for f in required_fields if not form_data.get(f)]
    
    if missing_fields:
        print(json.dumps({
            "success": False,
            "error": f"Missing required fields: {', '.join(missing_fields)}"
        }))
        return 1
    
    # Create HCP customer
    hcp_result = create_hcp_customer(form_data)
    
    # Save backup
    backup_file = save_lead_backup(form_data, hcp_result)
    
    # Return result
    result = {
        "success": hcp_result.get("success", False),
        "customer_id": hcp_result.get("customer_id"),
        "backup_file": backup_file,
        "message": "Lead captured successfully!" if hcp_result.get("success") else "Error capturing lead"
    }
    
    if not hcp_result.get("success"):
        result["error"] = hcp_result.get("error")
    
    print(json.dumps(result))
    return 0 if hcp_result.get("success") else 1

if __name__ == "__main__":
    sys.exit(main())
